13_0_Understanding_Your_Lightning_Setup.md

javiervargas opened the pull request:
I would like to get your comments and corrections.  Thanks.
J

shannona commented on 13_3_Setting_Up_a_Channel.md:
````diff
@@ -0,0 +1,260 @@
+
+```       
+$ lnd --listen=0.0.0.0:9736 
+```       
````
If it's really that easy to have LND on the same server as c-lightning, we can teach this all on one machine, but then we'd have to include the installation instructions for LND (rather than just depending on STandup).

shannona commented on 13_3_Setting_Up_a_Channel.md:
````diff
@@ -0,0 +1,260 @@
+
+```       
+$ lightning-cli --network=testnet connect 0302d48972ba7eef8b40696102ad114090fd4c146e381f18c7932a2a1d73566f84@127.0.0.1:9736
+{
````
Need to clarify where 0302d48972ba7eef8b40696102ad114090fd4c146e381f18c7932a2a1d73566f84 comes from. It might be that you move that getinfo command into the second of setting up the LND peer, so that you can have people set it up, then get the info, then come back and enter that into c-lightning.

Overall, we need a clear way to designate which commands are entered on which machine, possibly just:

cl$
lnd$

javiervargas replied:
Done,  I've included a paragraph explaining how to get pub key.   I've removed all LND commands.

shannona commented on 13_3_Setting_Up_a_Channel.md:
```diff
@@ -0,0 +1,260 @@
+
+#### Fund you c-lightning wallet.
+
+The first thing you need to do is send some satoshis to your c-lightning wallet.  You can create a new address using  `lightning-cli newaddr` command to use it later. The newaddr RPC command generates a new address which can subsequently be used to fund channels managed by the c-lightning node.  This last transaction is called the [funding transaction](https://github.com/lightningnetwork/lightnin ...
```
> :book: ***What is a c-lightning wallet?***

This is a great place to explain that it's a wallet that c-lightning will use to fund channels, not an actual channel.

javiervargas replied:
Done.

javiervargas commented on 13_1_Verifying_Your_Lightning_Setup.md:
````diff
@@ -0,0 +1,104 @@
+```
+"warning_bitcoind_sync": "Bitcoind is not up-to-date with network."
+```
+If you're not up-to-date, you'll get a message to you `getinfo command` like this:
````
Really it's local database c-lightning, but we could call it the lightning daemon in my opinion.

It seems confusing but if we can differentiate the commands as you have proposed I think it adds more value to have the two implementations together since it allows to exemplify the bitcoin transfer on each side. Additionally at BitcoinStandUp we are giving the option to choose any alternative even if the main implementation is c-lightning.

1.) I can add a specific section showing how to install LND.
2-3) I think your proposal c$ and lnd$ is awesome.
4) My intention is to use the LND commands to the minimum possible but if a few times to show the status of the channel on the other side and how money transfers are received.

javiervargas commented on 13_1_Verifying_Your_Lightning_Setup.md:
```diff
@@ -0,0 +1,104 @@
+> :information_source: **NOTE:** This is a draft in progress, so that I can get some feedback from early reviewers. It is not yet ready for learning.
+
+Before you start playing with c-lightning, you should ensure that everything is setup correctly.
+
```
Done

shannona replied:
Great.

javiervargas commented on 13_3_Setting_Up_a_Channel.md:
```diff
@@ -0,0 +1,260 @@
+#### Fund you c-lightning wallet.
+
+The first thing you need to do is send some satoshis to your c-lightning wallet.  You can create a new address using  `lightning-cli newaddr` command to use it later. The newaddr RPC command generates a new address which can subsequently be used to fund channels managed by the c-lightning node.  This last transaction is called the [funding transaction](https://github.com/lightningnetwork/lightnin ...
+
```
Corrected.

javiervargas commented on 13_3_Setting_Up_a_Channel.md:
````diff
@@ -0,0 +1,260 @@
+```       
+
+Now that we have funded our c-lightning wallet we will get information about remote node to start creating channel process.  On LND nodes you can get information about your node using `lncli -n testnet getinfo`:
+
````
You're right.  It should be only one,   I'll find a different way to get remote public key node and not use LND commands at all.
Commands are pretty different,  some a pair are the same casually.   I think we must renuncie show how invoices are payed between nodes.

javiervargas commented on 13_0_Understanding_Your_Lightning_Setup.md:
```diff
@@ -0,0 +1,33 @@
+
+For this and future chapters, we presume that you have a VPS with c-lightning installed, running `lightningd` courtersy of [Bitcoin Standup Scripts](https://github.com/BlockchainCommons/Bitcoin-Standup-Scripts).
+We also presume that you are connected to testnet, allowing for access to bitcoins without using real funds. You can either do this by hand, per [§2.1: Setting up a Bitcoin-Core VPS by Hand](02_1_Setting_Up_a_Bitcoin-Core_VPS_by_Hand.md), with Bitcoin Standup at Linode.com, per [§2.2: Setting up a Bitcoin-Core VPS with Bitcoin Standup](02_2_Setting_Up_a_Bitcoin-Core_VPS_with_StackScript.md), or v ...
+
```
I've talking with Jo about c-lightning installation on BitcoinStandup Script and it's basically the typical installation aside from linux service and standup user.   Do you think we can include c-lightning installation steps here?

shannona replied:
Basically, if it's in BitcoinStandup, I'd prefer to lean on that. Just edit §2.2 to tell the user what they have to do if they want to install Lightning, to use it down the road in this section.

But, there's probably no harm in briefly including the instructions here too (or rather in 13.1) if we're still kind of in an in-between state, where it might not be merged into both of the Standup repos yet. So, if that's still the case, go ahead and put it in, but be aware that it's something that may be removed down the line, so it doesn't need a whole lot of effort.

javiervargas replied:
Done, i've added c-lightning instructions.

> Thanks for the updates, Javier, they look great. Here's what I think we still need:
> 
> (1) Decide if you want to include the installation instruction, and if so they probably belong in 13.1
> (2) In 13.3, when we create the channel, we need to help the readers in the creation of a machine2 (or node2 or whatever) that they're working with, and then clearly display the commands on each server 'machine1-c$', 'machine2-c$' will match the way I've described multiple machines in previous chapters and match your 'c'.
> (3) We also need to describe how that invoice is being created, presumably on the other machine, probably in a new section in advance of the current 13.4.

In 13.3 we're using only c-lightning commands,  no more lnd commands.    I'm updating 13.4 to show lnd commands clarifying which are lnd commands or c-lightning commands.

javiervargas commented on 13_3_Setting_Up_a_Channel.md:
```diff
@@ -0,0 +1,260 @@
+In simple terms a lightning channel is a money tube that allows fast, cheap and private transfers of money without sending transactions to the blockchain. 
+More technically a channel is a 2-of-2 multisignature bitcoin onchain transaction that establishes a financial relationship between two agents without trust.
+As we seen in [§6.1:Sending a transaction to a multisig](06_1_Sending_a_Transaction_to_a_Multisig.md) chapter multisignatures are generally described as being "m of n". That means that the transaction is locked with a group of "n" keys, but only "m" of them are required to unlock the transaction.   
+
```
Done,  take a look.

shannona replied:
Good focus.

shannona commented on 13_3_Setting_Up_a_Channel.md:
```diff
@@ -0,0 +1,230 @@
+The first thing you need to do is connect your node to a peer. This is done with the `lightning-cli connect` command. Remember that if you want more information on this command, you should type `lightning-cli help connect`.   The connect RPC command establishes a new connection with another node in the Lightning Network.
+
+To connect your node to a remote peer you need it's id that represents the target node’s public key. As a convenience, id may be of the form id@host or id@host:port.  Using `lightning-cli listnodes` command you obtain all nodes available on the network and choose one.
+
```
We do need to tell people how to set up that remote node, however they do it. The goal with these tutorials is always to present the real opportunity to work through everything and try out all the commands, and here we have a gap. So it might be as simple as saying "set up a Lightning node on a second machine using the Standup script". I just want to make it clear for people that it's a step.

javiervargas replied:
Done.

shannona commented on 13_4_Paying_a_Invoice.md:
```diff
@@ -0,0 +1,146 @@
+Consider this ln invoice:
+
+`lntb1m1p03ft7lpp5zve4dsgwgdxekqqq39vhgcnv6gfa2g2ktqy9lf0aph60d0388xmqdqqcqzpgsp545a9fphd8m5ayplcu8m5845cr4m0zcnyxddwv4g3zm32yprkfd4q9qy9qsq3s4y6cmyvh0qw9qm0sf80llxyyjy9xwrjds7lpkqhzv247jsm6q5me8t9e6ftquma664gz5u4a2rvs0yf4f0mlwtwfs6as5uj5djzhcqpnqlcj
+`  
```
What we need to say here is where that invoice came from. That can be: we'll see how to generate the invoice in the next chapter.  But how I originally had it laid out was (1) create the invoice in one section; (2) then pay it in the next. That offers the advantage of again this being a real-world example, so someone can see how to generate the invoice on one machine, then pay it on the other.

javiervargas replied:
I've ordered chapters to create first and then pay it

shannona commented:
Thanks for the updates, Javier, they look great. Here's what I think we still need:

(1) Decide if you want to include the installation instruction, and if so they probably belong in 13.1
(2) In 13.3, when we create the channel, we need to help the readers in the creation of a machine2 (or node2 or whatever) that they're working with, and then clearly display the commands on each server 'machine1-c$', 'machine2-c$' will match the way I've described multiple machines in previous chapters and match your 'c'.
(3) We also need to describe how that invoice is being created, presumably on the other machine, probably in a new section in advance of the current 13.4.

shannona commented:
A bit more future-looking:

(4) After all that, do you *want* to have a separate section on LND that quickly runs through the same commands? Your call as to whether you think it's helpful, but that'd be the way to separate it and so not cause confusion. (If so you could include a c-lightning or LND section that talks about the advantages of both.)
(5) After all of that, are there any other major CLI commands we should teach for lightning?

shannona commented:
Oh, I do see in my outline regarding (5) that I have at least one more thing flagged: closing a channel (and seeing how your money comes out). The question is if there's anything before that.

javiervargas commented:
> Thanks for the updates, Javier, they look great. Here's what I think we still need:
> 
> (1) Decide if you want to include the installation instruction, and if so they probably belong in 13.1
> (2) In 13.3, when we create the channel, we need to help the readers in the creation of a machine2 (or node2 or whatever) that they're working with, and then clearly display the commands on each server 'machine1-c$', 'machine2-c$' will match the way I've described multiple machines in previous chapters and match your 'c'.
> (3) We also need to describe how that invoice is being created, presumably on the other machine, probably in a new section in advance of the current 13.4.

(1) I've added installation instructions.
(2) I've fixed order chapters to follow correct flow about create invoice and later pay it.   I've indicated 'lnd$' and 'c$' shells.

javiervargas commented:
> A bit more future-looking:
> 
> (4) After all that, do you _want_ to have a separate section on LND that quickly runs through the same commands? Your call as to whether you think it's helpful, but that'd be the way to separate it and so not cause confusion. (If so you could include a c-lightning or LND section that talks about the advantages of both.)
> (5) After all of that, are there any other major CLI commands we should teach for lightning?

(4) I was planning to show how money flows between nodes after payment and later close channel seeing final state each side.
Ok,  I'll add a advantages sections each one.
(5)  IMO yes,  We could show all list 'invoices','peers','nodes','transactions' commands.   There are deeper ones but they are probably out of reach

javiervargas commented on 13_3_Setting_Up_a_Channel.md:
````diff
@@ -0,0 +1,260 @@
+
+```       
+$ lncli -n testnet getinfo
+{
````
Just to clarify, lncli is not the alias but the actual LND command

shannona commented on 13_5_Paying_a_Invoice.md:
```diff
@@ -0,0 +1,151 @@
+Check payee value is the same as identity_pubkey value seen previous chapter.
+
+**"identity_pubkey": "0302d48972ba7eef8b40696102ad114090fd4c146e381f18c7932a2a1d73566f84",**
+
```
I note this is the same peer that we set the channel up with. Which is expected.

It might be worth a bit of commentary on this, also: (1) can people expect to see payment requests from people further out on the Lightning Network (obviously, that's in the spec, I just don't know if it's reality yet); and (2) if so do they need to do anything special to pay them. (I assume not; but if that's the case it'd be good to just mention that all.)

shannona commented on 13_6_Closing_a_Channel.md:
````diff
@@ -0,0 +1,210 @@
+   ]
+}
+```
+
````
So a channel is identified by its peer's id? That's what it looks like from your further examples. If so, let's say it.

javiervargas replied:
Actually there is a channel id that identifies a channel

`"channels": [
      {
         "peer_id": "025e391e219227d2b795c884ea34852c8d1995ba095671a7240915d07b3a984836",
         "connected": false,
         "state": "CHANNELD_NORMAL",
         "short_channel_id": "1803856x5x0",
         "channel_sat": 10,
         "our_amount_msat": "10000msat",
         "channel_total_sat": 100000,
         "amount_msat": "100000000msat",
         "funding_txid": "ba7cf2118c63496b10ab89abe6533f1302846ba42eadfe93cc41607e721f09a1",
         "funding_output": 0
      }
   ]
`
"short_channel_id": "1803856x5x0" means a tx created on 1803856 block,  tx index (5), output index (0) for that channel.  I'll add it.

shannona commented on 13_6_Closing_a_Channel.md:
```diff
@@ -0,0 +1,210 @@
+
+In this case when only one participant is online or if the participants disagree on the last state of the channel,  so one peer can perform an unilateral close of the channel without the cooperation of the other node.   It's performed by broadcasting a commitment transaction that commits to a previous channel state which both parts have agreed upon.
+This commitment transaction contains the channel state divided in two parts: the balance of each participant and all the pending payments (HTLCs).
+
```
Do these differences require anything different in the close command? If not, we should tightly summarize them. (Is the close command below an example of a cooperative close, a force close, or a cooperative close that will become a force close if there's no response? We just need to link up this theory to the practice.)

javiervargas replied:
Yes, i've added information about how to perform this kind of clousure.

shannona commented on 13_6_Closing_a_Channel.md:
````diff
@@ -0,0 +1,210 @@
+Finally you should use `lightning-cli close` command to close the channel. The close RPC command attempts to close the channel cooperatively with the peer,  if you want to close it unilaterally set unilateraltimeout argument with number of seconds command will wait.   If you set to 0 and the peer is online command can negotiate a mutual close.   For this example we use an mutual close.
+
+```
+lightning-cli --network=testnet close 0302d48972ba7eef8b40696102ad114090fd4c146e381f18c7932a2a1d73566f84 0
````
General practice is to capture that long ID if possible. Is it always the destination? If so, we should say that and capture it. (We can always show the whole info, like you do above, then afterward show a capture of the required variable.)

shannona commented on 13_6_Closing_a_Channel.md:
```diff
@@ -0,0 +1,210 @@
+The closing transaction onchain is [b4c0a1993dd113081eff5369a22d6afe1af9f0d07b29a590e8772ac7f712736a](https://blockstream.info/testnet/tx/b4c0a1993dd113081eff5369a22d6afe1af9f0d07b29a590e8772ac7f712736a).
+
+This transaction has two outputs, one for remote node and other for local c-lightning wallet.   Output on index 0 corresponds to remote node with a value of 100000.   Output on index 1 correspond to local node with a value of 179817.
+
```
So the user doesn't need to do anything? The close will generate the transaction?

javiervargas replied:
Yes,  it does.

shannona commented on 13_7_Lightning_Network_Implementations.md:
```diff
@@ -0,0 +1,5 @@
+| Name  | Description | 
+| ------------- | ------------- |
```
I think this is a good section to have, but you can keep it very brief. Just have a short section with how to download that one (and/or note if it's an option in Standup), and its advantages and disadvantages.

shannona commented:
Please rename files per the following table of contents:

* 18.0: Understanding Lightning
  * 18.1: Verifying Your Lightning Setup
  * 18.2: Knowing Your Lightning Setup
  * 18.3: Setting up a Channel
* 19.0: Using Lightning
  * 19.1: Generating a Payment Request
  * 19.2: Paying an Invoice
  * 19.3: Closing a Channel

shannona commented:
Great Stuff!
